From c3e566d59712c7460a1055a7f1071803c60cae26 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 26 Jul 2022 16:10:31 +0300
Subject: [PATCH] [SW-97612] gaudi2: override the default prot mask

The default protection mask specifies protection of the whole flash.
Gaudi2 needs to protect only the eROM binary area which is 128KB from
the start of the flash.
Override the PROT_MASK with Gaudi2 specific value.

Change-Id: I5533783a808a7c1f6237199946d58317a272f320
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 8c5ca92..cbf43cb 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -118,6 +118,15 @@
 #define TS_CALIB_B_H_COEFF	262600
 
 #ifndef __ASSEMBLY__
+#include <spi_flash.h>
+
+/* Let this file be included in any order and override the PROT_MASK */
+#ifdef PROT_MASK
+#undef PROT_MASK
+#endif
+/* Only BP1 == 0x2 - means 2 blocks - up to 128KB (eROM) */
+#define PROT_MASK	(READ_STATUS_BP1 | READ_STATUS_SRWD)
+
 #define sleep()	__asm__ __volatile__("sleep\n\t" : : )
 
 void init_secondary(void);
